From 3247690bbc075d37863f6f2c9f8f33c5ee6276ee Mon Sep 17 00:00:00 2001
From: Jared Smudde <computerwhiz02@hotmail.com>
Date: Thu, 4 Jun 2015 14:01:14 -0500
Subject: inetcpl: Implement default page button.

This patch implements the button default page to set back the
default homepage is the user changes the homepage.

 dlls/inetcpl.cpl/general.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)
---
 dlls/inetcpl.cpl/general.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/dlls/inetcpl.cpl/general.c b/dlls/inetcpl.cpl/general.c
index 0691579..24631c6 100644
--- a/dlls/inetcpl.cpl/general.c
+++ b/dlls/inetcpl.cpl/general.c
@@ -35,6 +35,7 @@
 WINE_DEFAULT_DEBUG_CHANNEL(inetcpl);
 
 static const WCHAR about_blank[] = {'a','b','o','u','t',':','b','l','a','n','k',0};
+static const WCHAR default_home[] = {'h','t','t','p',':','/','/','w','w','w','.','w','i','n','e','h','q','.','o','r','g',0};
 static const WCHAR start_page[] = {'S','t','a','r','t',' ','P','a','g','e',0};
 static const WCHAR reg_ie_main[] = {'S','o','f','t','w','a','r','e','\\',
                                     'M','i','c','r','o','s','o','f','t','\\',
@@ -43,7 +44,6 @@ static const WCHAR reg_ie_main[] = {'S','o','f','t','w','a','r','e','\\',
 
 /* list of unimplemented buttons */
 static DWORD disabled_general_buttons[] = {IDC_HOME_CURRENT,
-                                           IDC_HOME_DEFAULT,
                                            IDC_HISTORY_SETTINGS,
                                            0};
 static DWORD disabled_delhist_buttons[] = {IDC_DELETE_FORM_DATA,
@@ -187,6 +187,10 @@ static INT_PTR general_on_command(HWND hwnd, WPARAM wparam)
             SetDlgItemTextW(hwnd, IDC_HOME_EDIT, about_blank);
             break;
 
+        case MAKEWPARAM(IDC_HOME_DEFAULT, BN_CLICKED):
+            SetDlgItemTextW(hwnd, IDC_HOME_EDIT, default_home);
+            break;
+
         case MAKEWPARAM(IDC_HISTORY_DELETE, BN_CLICKED):
             DialogBoxW(hcpl, MAKEINTRESOURCEW(IDD_DELETE_HISTORY), hwnd,
                        delhist_dlgproc);
-- 
2.4.2

